Proguard preventing build using JDK 17 (Cannot support JDK 8 and 17)

**Problem**

Using JDK `java 17.0.1 2021-10-19 LTS` and Gradle `7.3`

When I run `./gradlew build` I get the following...

```
> Task :proguardJar FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':proguardJar'.
> java.io.IOException: Can't read [/home/theman/Applications/Java/jdk-17.0.1/jmods] (Can't read [jdk.jstatd.jmod] (Can't process class [module-info.class] (Unsupported version number [61.0] (maximum 60.65535, Java 16))))

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org
```

It seems proguard should support this at least in Beta (https://github.com/Guardsquare/proguard/issues/132) so it should probably be upgraded to support 17 builds.

**Current workaround** 

I am guessing I can use the 16 JDK but I don't have time to try that rt now. I will setup a diff JDK shortly to confirm.

Sorry if I did this wrong new to contributing here.

**Per @modmuss50 we cannot just upgrade proguard**

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -4,7 +4,7 @@ jobs:
   build:
     strategy:
       matrix:
-        java: [8-jdk, 11-jdk, 16-jdk]
+        java: [8-jdk, 11-jdk, 16-jdk, 17-jdk]
     runs-on: ubuntu-20.04
     container:
       image: eclipse-temurin:${{ matrix.java }}

--- a/.github/workflows/release.yml
+++ b/.github/workflows/release.yml
@@ -4,7 +4,7 @@ jobs:
   build:
     runs-on: ubuntu-20.04
     container:
-      image: eclipse-temurin:16-jdk
+      image: eclipse-temurin:17-jdk
       options: --user root
     steps:
       - run: apt update && apt install git -y && git --version

--- a/build.gradle
+++ b/build.gradle
@@ -1,7 +1,7 @@
 buildscript {
 	dependencies {
 		classpath 'org.kohsuke:github-api:1.114'
-		classpath 'com.guardsquare:proguard-gradle:7.1.0'
+		classpath 'com.guardsquare:proguard-gradle:' + (JavaVersion.current().isCompatibleWith(JavaVersion.VERSION_11) ? '7.2.0-beta2' : '7.1.0')
 	}
 }
 

